stdlib/number.gr

```diff
@@ -135,3 +137,61 @@ export let abs = (x: Number) => if (0 > x) x * -1 else x
  * @returns Number
  */
 export let neg = (x: Number) => if (x > 0) x * -1 else x
+
+/**
+ * Checks if a number is finite.
+ * All values are finite exept for floating point NaN, infinity or negative infinity.
+ * @param x: Number - The number to check
+ * @returns Bool
+ */
+@disableGC
+export let isFinite = (x: Number) => {
+	let asPtr = WasmI32.fromGrain(x)
+	if (isBoxedNumber(asPtr)) {
+		// Boxed numbers can have multiple subtypes, Of which float32 and float64 can be infinite,
+		let tag = WasmI32.load(asPtr, 4n)
+		if (WasmI32.eq(tag, Tags._GRAIN_FLOAT64_BOXED_NUM_TAG)) {
+			let wf64 = WasmF64.load(asPtr, 8n)
+			WasmF64.eq(WasmF64.sub(wf64, wf64), 0.W)
```

Comment:
This comment might help with readability:
```suggestion
			// uses the fact that all finite floats minus themselves are zero
			// (NaN - NaN == NaN, inf - inf == NaN, 
			//  -inf - -inf == NaN, inf - -inf == inf, -inf - inf == -inf)
			WasmF64.eq(WasmF64.sub(wf64, wf64), 0.W)
```

Reply:
Added the comment.